### ISSUE TITLE
initializeDataSource.py dies if the --ds_foldername is an absolute path.

### ISSUE THREAD

**Issue by alanhoyle** [broadinstitute/oncotator#350]
I'm trying to create a custom data source using an ExAC VCF file.  I'm running oncotator 1.9.0 in a Docker instance created an image made using the shipped Dockerfile.  

If I set the --ds_foldername to an absolute path (e.g. "--ds_foldername=/db_dir/exac"), it dies after it builds the index and gzip parts as it uses the path to create a temporary build before copying it into place.  When it tries to copy the results into place, it can't create the absolute path, as it already exists, then just dies.  

work-around:  if I put "--ds_foldername=exac --dbDir=/db_dir" it works fine:  it puts the build files in a temporary location and then creates /db_dir/exac with the expected output at the end.  

Aside:  it might be worth documenting how to do build new Data Sources from the Docker....

**Comment by LeeTL1220** [comment]
Yes, (apparently) we did not document initialize datasource well enough.
The ds_foldername is meant to be the name of the folder created inside the
dbdir.

Also, exac is in the next default datasource corpus, which I would have
uploaded by now, except I've been busy with AACR.

**Comment by LeeTL1220** [comment]
Just for you (and anyone else reading this)...  I had a hiccup with the
forums, which are being fixed.  So these are the updated links until that
is sorted

Default datasources:
http://www.broadinstitute.org/~lichtens/oncobeta/oncotator_v1_ds_April052016.tar.gz

v1.9.0.0:
https://github.com/broadinstitute/oncotator/archive/v1.9.0.0.tar.gz

On Mon, Apr 18, 2016 at 7:16 PM, Lee Lichtenstein <
lichtens@broadinstitute.org> wrote:

> Yes, (apparently) we did not document initialize datasource well enough.
> The ds_foldername is meant to be the name of the folder created inside the
> dbdir.
>
> Also, exac is in the next default datasource corpus, which I would have
> uploaded by now, except I've been busy with AACR.
> On Apr 18, 2016 6:04 PM, "Alan Hoyle" <notifications@github.com> wrote:
>
>> I'm trying to create a custom data source using an ExAC VCF file. I'm
>> running oncotator 1.9.0 in a Docker instance created an image made using
>> the shipped Dockerfile.
>>
>> If I set the --ds_foldername to an absolute path (e.g.
>> "--ds_foldername=/db_dir/exac"), it dies after it builds the index and gzip
>> parts as it uses the path to create a temporary build before copying it
>> into place. When it tries to copy the results into place, it can't create
>> the absolute path, as it already exists, then just dies.
>>
>> work-around: if I put "--ds_foldername=exac --dbDir=/db_dir" it works
>> fine: it puts the build files in a temporary location and then creates
>> /db_dir/exac with the expected output at the end.
>>
>> Aside: it might be worth documenting how to do build new Data Sources
>> from the Docker....
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/broadinstitute/oncotator/issues/350>
>>
>


-- 
Lee Lichtenstein
Broad Institute
75 Ames Street, Room 7003EB
Cambridge, MA 02142
617 714 8632
